src/Bicep.LangServer/Handlers/BicepBuildCommandHandler.cs

```diff
@@ -57,21 +57,21 @@ public override Task<string> Handle(string bicepFilePath, CancellationToken canc
 
         private string GenerateCompiledFileAndReturnBuildOutputMessage(string bicepFilePath, DocumentUri documentUri)
         {
+            CompilationContext? context = compilationManager.GetCompilation(documentUri);
+
+            if (context is null)
+            {
+                return $"Build failed. The file \"{bicepFilePath}\" may not be a Bicep file and is not tracked by the Bicep language server.";
```

Comment:
I did some experimenting with the existing behavior, and it led me to wonder if we really should be relying on the context being populated in advance by the language server in this handler.

I can create a new bug for this as it's not strictly related to the origin issue, but basically if I right-click + Build on any `.bicep` file that **isn't** open in my editor, the command fails (I assume because of the context lookup).

Here I do think it might be good to continue throwing the exception until we come up with a solution to that bug, as it gives a more obvious visual feedback that something went wrong. Thoughts?

Reply:
Wow I was not aware of that bug. Creating a new bug sounds good to me, and it makes sense to revert my changes since the error message may be even more confusing due to the fact that we are not able to build a valid Bicep file because it "may not be a Bicep file".

Reply:
Reverted the changes.

Reply:
Sounds good - I've created #4287